src/ray/core_worker/object_interface.h

```diff
@@ -94,6 +106,9 @@ class CoreWorkerObjectInterface {
   /// Store socket name.
   std::string store_socket_;
 
+  /// In-memory store for return objects. This is used for `MEMORY` store provider.
+  std::shared_ptr<CoreWorkerMemoryStore> memory_store_;
```

Comment:
Why do we need this reference? Shouldn't it be owned by the memory store provider?

Reply:
Good question. This is because store provider is basically a store client, if I create two memory store providers in a worker, they should point to the same memory store, instead of creating two stores. This is the same for plasma store provider as well.